modules/account-lib/src/coin/dot/keyPair.ts

```diff
@@ -12,18 +19,56 @@ export class KeyPair extends Ed25519KeyPair {
     super(source);
   }
 
+  /**
+   * Helper function to create the KeyringPair for signing a dot transaction.
+   *
+   * @returns {KeyringPair} dot KeyringPair
+   * @see https://polkadot.js.org/docs/api/start/keyring
+   */
+  protected createPolkadotPair(): KeyringPair {
+    const secretKey = this.keyPair.prv ? new Uint8Array(Buffer.from(this.keyPair.prv, 'hex')) : new Uint8Array();
```

Comment:
Is `new Uint8Array()` going to provide the same array over and over again? Why?

Reply:
Ideally it should be random

Reply:
The second path is only taken incase we cant to create a keypair when with only a public key. The secret key wont matter in that case, sort of what we do in the ed25519KeyPair class on line 31

Reply:
Then the prv should be `undefined`, we shouldn't give the user a generic key.

> sort of what we do in the ed25519KeyPair class on line 31 
https://github.com/BitGo/BitGoJS/blob/b6090cab84f867037d8f40267344426dfcc56244/modules/account-lib/src/coin/baseCoin/ed25519KeyPair.ts#L31
Not sure I follow